subprojects/version-control/src/main/java/org/gradle/vcs/VersionControlSpec.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.vcs;
+
+import java.net.URI;
+
+/**
+ * Captures user-provided information about a version control system.
+ */
+public interface VersionControlSpec {
```

Comment:
We can probably leave this just as an empty interface.  Most version control systems will have a URL like thing, but I don't think we have to require it.  

Maybe this should be `Describable`?